recipes/makefile-project-workspace-creator/all/conanfile.py

```diff
@@ -1,37 +1,55 @@
 import os
-from conans import ConanFile, tools
+
+from conan import ConanFile, conan_version
+from conan.tools.files import copy, get
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.47.0"
 
 
 class MPCGeneratorConan(ConanFile):
     name = "makefile-project-workspace-creator"
     description = "The Makefile, Project and Workspace Creator"
     license = "BSD-3-Clause"
-    homepage = "https://objectcomputing.com/"
     url = "https://github.com/conan-io/conan-center-index"
-    topics = ("conan", "makefile-project-workspace-creator", "objectcomputing", "installer")
-    settings = "os"
+    homepage = "https://github.com/objectcomputing/MPC"
+    topics = ("objectcomputing", "installer")
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    package_type = "build-scripts"
+    settings = "os", "arch", "compiler", "build_type"
 
-    def requirements(self):
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def build_requirements(self):
         if self.settings.os == "Windows":
-            self.requires("strawberryperl/5.30.0.1")
+            kwargs = dict(visible=True) if conan_version.major >= 2 else {}
+            self.tool_requires("strawberryperl/5.32.1.1", **kwargs)
 
-    def build(self):
-        pass
+    def package_id(self):
+        del self.info.settings.compiler
```

Comment:
Why does `arch` should be considered now? It's a bunch of Perl scripts only.

Reply:
I agree. Replaced it with `self.info.clear()`.